components/homme/src/theta-l_kokkos/cxx/HyperviscosityFunctorImpl.hpp

```diff
@@ -149,7 +150,42 @@ class HyperviscosityFunctorImpl
         });
       }
 #endif
-    });
+    }); //team thread range
+
+    // Laplacian of layer thickness
+    m_sphere_ops.laplace_simple(kv,
```

Comment:
I _think_ you might need a team barrier before this point, since you must ensure the subtraction of the ref states is complete before you attempt to take its grad inside the laplace simple routine. Or, better, you could start with the laplace of something not involved in the ref states above, like the lapl of V, so we save ourselves a barrier.

Reply:
Great, thanks, will do.

Reply:
the code that shows as new on GitHub, girstLaplaceHV, is old, but I put a team barrier there, too (as well as in nutopLaplace), thanks.